plugins/plugin-java/che-plugin-java-server/src/main/java/org/eclipse/che/plugin/java/languageserver/JavaLanguageServerLauncher.java

```diff
@@ -94,11 +99,27 @@ public JavaLanguageServerLauncher(
     this.eventService = eventService;
     this.projectManager = projectManager;
     this.projectSynchronizer = projectSynchronizer;
+    this.notificationHandler =
+        Executors.newSingleThreadScheduledExecutor(
+            new ThreadFactoryBuilder()
+                .setNameFormat("Jdtls Notification Handler")
+                .setUncaughtExceptionHandler(LoggingUncaughtExceptionHandler.getInstance())
+                .setDaemon(true)
+                .build());
 
     launchScript = Paths.get(System.getenv("HOME"), "che/ls-java/launch.sh");
   }
 
+  @PreDestroy
+  public void shutDown() {
+    notificationHandler.shutdownNow();
```

Comment:
About this line: we don't want any notifications to be processed after this line, so "shutdownNow" is the right thing to do. We don't need to wait for termination here: if a notification is being processed currently, it will broken out of any waits by thread interruption. If the notification is hung, no amount of waiting will make it terminate. If it is not hung, it will soon be done and the executor thread will terminate.